sdk/search/azure-search/azure/search/_index/_search_index_client.py

```diff
@@ -0,0 +1,329 @@
+            key=key, selected_fields=selected_fields, **kwargs
+        )
+        return cast(dict, result.additional_properties)
+
+    @distributed_trace
+    def search(self, query, **kwargs):
+        # type: (Union[str, SearchQuery], **Any) -> ItemPaged[dict]
+        """Search the Azure search index for documents.
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sample_simple_query.py
+                :start-after: [START simple_query]
+                :end-before: [END simple_query]
+                :language: python
+                :dedent: 4
+                :caption: Search on a simple text term.
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sample_filter_query.py
+                :start-after: [START filter_query]
+                :end-before: [END filter_query]
+                :language: python
+                :dedent: 4
+                :caption: Filter and sort search results.
+        """
+        if isinstance(query, six.string_types):
+            query = SearchQuery(search_text=query)
+        elif not isinstance(query, SearchQuery):
+            raise TypeError(
+                "Expected a string or SearchQuery for 'query', but got {}".format(
+                    repr(query)
+                )
+            )
+
+        return ItemPaged(
+            self._client, query, kwargs, page_iterator_class=_SearchDocumentsPaged
+        )
+
+    @distributed_trace
+    def suggest(self, query, **kwargs):
+        # type: (Union[str, SuggestQuery], **Any) -> List[dict]
+        """Get search suggestion results from the Azure search index.
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sample_suggestions.py
+                :start-after: [START suggest_query]
+                :end-before: [END suggest_query]
+                :language: python
+                :dedent: 4
+                :caption: Get search suggestions.
+        """
+        if not isinstance(query, SuggestQuery):
+            raise TypeError(
+                "Expected a SuggestQuery for 'query', but got {}".format(repr(query))
+            )
+
+        response = self._client.documents.suggest_post(
+            suggest_request=query.request, **kwargs
+        )
+        results = [r.as_dict() for r in response.results]
+        return results
+
+    @distributed_trace
+    def autocomplete(self, query, **kwargs):
+        # type: (Union[str, AutocompleteQuery], **Any) -> List[dict]
+        """Get search auto-completion results from the Azure search index.
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sample_autocomplete.py
+                :start-after: [START autocomplete_query]
+                :end-before: [END autocomplete_query]
+                :language: python
+                :dedent: 4
+                :caption: Get a auto-completions.
+        """
+        if not isinstance(query, AutocompleteQuery):
```

Comment:
We should accept `str` like objects as well, no?

Reply:
Two arguments are mandatory, the search text and a suggester. I'd personally prefer to avoid the complication of dealing with combinations that don't make sense.